ddtrace/tracer/metrics_test.go

```diff
@@ -46,20 +44,23 @@ func TestReportMetrics(t *testing.T) {
 }
 
 type testGauger struct {
-	mu    sync.RWMutex
-	calls []string
-	tags  []string
-	ch    chan<- struct{}
+	mu     sync.RWMutex
+	calls  []string
+	tags   []string
+	waitCh chan struct{}
+	n      int
 }
 
 func (tg *testGauger) Gauge(name string, value float64, tags []string, rate float64) error {
 	tg.mu.Lock()
 	defer tg.mu.Unlock()
 	tg.calls = append(tg.calls, name)
 	tg.tags = tags
-	select {
-	case tg.ch <- struct{}{}:
-	default:
+	if tg.n > 0 {
```

Comment:
Why not just have a local `n` in `Wait` and keep decreasing that? Doesn't that make more sense? It will be local to the call.